lua/pug/modules/lagdetection.lua

```diff
@@ -95,28 +109,35 @@ u.addHook("Tick", "LagDetection", function()
 
 	if not inTolerance then
 		skips = skips + 1
-		if ( lag.rate > lag.trigger ) or ( skips > lag.skips ) then
-			ServerLog( "LAG: ", lag.rate, "Th: ", lag.threshold )
-			ServerLog( "Average: ", sample.mean, "Skips: ",
-			skips, "/", lag.skips )
+		print("Skips: " .. tostring(skips))
+		print("Rate: " .. tostring(lag.rate))
+		print("Average: " .. tostring(sample.mean))
+
+		-- IF there's casual lag
+		if ( sample.mean <= lag.trigger ) or ( skips >= lag.skips ) then
+			print( "LAG!", "Rate: ", lag.rate, "Trigger: ", lag.trigger )
+			print( "Average: ", sample.mean, "Skips: ", skips, "/", lag.skips ) 
 
 			lag.fClean()
 			lag.timeout = sysTime + lag.cooldown
 
 			PUG:Notify( "pug_lagdetected", 3, 5, nil )
 
-			if ( lag.rate > lag.panic ) or ( skips > lag.pSkips )  then
-				lag.fPanic()
-				PUG:Notify( "pug_lagpanic", 3, 5, nil )
-			end
+        end
+
+        -- If there's a panic
+        if ( sample.mean <= lag.panic ) or ( skips >= lag.pSkips )  then
```

Comment:
Same comment as the previous check, but we also moved this check to be a sibling of the other one, rather than a child.

This way we could technically have whatever values we wanted without requiring that a normal cleanup occur before a panic. Why would you want to do this? I'm not sure, but I can't see a reason not to allow it.

Reply:
That doesn't really make sense as the `panic` value should always be greater then the regular `detection` value. Putting it inside the statement was just a way to fool-proof the settings, so that people don't get it working backwards.